h2o-clustering/src/main/java/water/clustering/AssistedClusteringEmbeddedConfigProvider.java

```diff
@@ -0,0 +1,79 @@
+package water.clustering;
+
+import org.apache.log4j.Logger;
+import water.clustering.api.AssistedClusteringRestApi;
+import water.clustering.api.GracefulAsyncRunner;
+import water.init.AbstractEmbeddedH2OConfig;
+import water.init.EmbeddedConfigProvider;
+import water.util.Log;
+
+import java.io.IOException;
+import java.util.Optional;
+import java.util.function.Consumer;
+
+/**
+ * Provides {@link AssistedClusteringEmbeddedConfig} as a result of external assist. The resulting
+ * embedded config provides a flatfile assembled by external service, given to H2O. 
+ * This EmbeddedConfig has no timeout and will wait indefinitely until a flatfifle is received.
+ */
+public class AssistedClusteringEmbeddedConfigProvider implements EmbeddedConfigProvider {
+    private static final Logger LOG = Logger.getLogger(AssistedClusteringEmbeddedConfigProvider.class);
+
+    private AssistedClusteringEmbeddedConfig assistedClusteringEmbeddedConfig;
+    private String flatfile;
+    private final Object notification = new Object();
+
+    @Override
+    public void init() {
+        final Consumer<String> flatFileCallback = s -> {
```

Comment:
This method is called to initialize the provider, regardless of if the provider is active or not, see javadoc:

    /**
     * Provider initialization. Guaranteed to be called before any other method is called, including the`isActive`
     * method.
     */

This means you shouldn't probably do anything like this in this method, instead, you would want to move this code to `getConfig`

Reply:
Moved.